src/backend/utils/highlighter.ts

```diff
@@ -66,6 +66,8 @@ export const highlight = (nodes: Array<Node>) => {
     clear();
   }
 
+  nodes = nodes.filter(n => n != null);
```

Comment:
Since this is all in scope, maybe it would be fine to just use it in the loop at line 75.